### ISSUE TITLE
Intermittent network errors connecting dev db server ( A transport-level error has occurred when receiving results from the server )

### ISSUE THREAD

**Issue by jdhillon123** [dotnet/SqlClient#2621]
024-06-21T15:11:50.9938424Z /p:importFile=Import2_04_TR_TRP.prt;logFile="C:\ProgramData\Blob Technologies\project\7.1\Import2_04_TR_TRP_20240621_111010.log"
2024-06-21T15:11:50.9939041Z /c:server=projectdevdb2\sql2017;database=project_71_BVT_import;user=project;password=password
2024-06-21T15:11:50.9939618Z Project.Console Information: 0 : ======================= prtConsole 7.1.7.1925 =======================
2024-06-21T15:11:50.9940193Z Project.Console Information: 0 : prtConsole tracing started at 6/21/2024 11:10:10 AM
2024-06-21T15:11:50.9940737Z Project.Console Information: 0 : Successful login for user TFSBUILDSERVICE.
2024-06-21T15:11:50.9941831Z Project.Console Information: 0 : Properties: /service:import /target:load /connectionstring:server=projectdevdb2\sql2017;database=project_71_BVT_import;user=project;password=****** /property:importfile=Import2_04_TR_TRP.prt;logfile=C:\ProgramData\COMPANY Technologies\Project\7.1\Import2_04_TR_TRP_20240621_111010.log; /log:event=False;console=True;file=;level=Information; 
2024-06-21T15:11:50.9943017Z Project.Console Information: 0 : Generated session id chb45aes-34334-527j-5155-e23658db1061 as sessionid parameter was not supplied.
2024-06-21T15:11:50.9943758Z Project.Console Information: 0 : Optional dumpfile parameter was not supplied. Using C:\Tests\BuildValueTests\Import2_04_TR_TRP.dmp.
2024-06-21T15:11:50.9944425Z Project.Console Information: 0 : Creating session chb45aes-34334-527j-5155-e23658db1061.
2024-06-21T15:11:50.9945145Z Project.Console Information: 0 : Starting Import session chb45aes-34334-527j-5155-e23658db1061 for file: C:\Tests\BuildValueTests\Import2_04_TR_TRP.prt
2024-06-21T15:11:50.9945972Z Project.Console Information: 0 : Updating status of session chb45aes-34334-527j-5155-e23658db1061 to Active
2024-06-21T15:11:50.9946632Z Project.Console Information: 0 : Initializing dump file C:\Tests\BuildValueTests\Import2_04_TR_TRP.dmp
2024-06-21T15:11:50.9947087Z Project.Console Information: 0 : Starting bulk copy into staging tables...
2024-06-21T15:11:50.9947495Z Project.Console Information: 0 : Initializing handler for TRP record type.
2024-06-21T15:11:50.9947895Z Project.Console Information: 0 : Initializing handler for TR record type.
2024-06-21T15:11:50.9948299Z Project.Console Information: 0 : Writing 13 TRP records to staging table
2024-06-21T15:11:50.9948691Z Project.Console Information: 0 : Writing 23 TR records to staging table
2024-06-21T15:11:50.9949082Z Project.Console Information: 0 : Bulk Import processor disposed.
2024-06-21T15:11:50.9949465Z Project.Console Information: 0 : Starting validation for 13 TRP records
2024-06-21T15:11:50.9949856Z Project.Console Information: 0 : 0 records remaining to validate
2024-06-21T15:11:50.9950214Z Project.Console Information: 0 : Saving 13 TRP records
2024-06-21T15:11:50.9951548Z Project.Console Error: 0 : Error encountered saving TRP records. System.Data.SqlClient.SqlException (0x80131904): A transport-level error has occurred when receiving results from the server. (provider: TCP Provider, error: 0 - The specified network name is no longer available.) ---> System.ComponentModel.Win32Exception (0x80004005): The specified network name is no longer available
2024-06-21T15:11:50.9954000Z    at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
2024-06-21T15:11:50.9954605Z    at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
2024-06-21T15:11:50.9955189Z    at System.Data.SqlClient.TdsParserStateObject.ReadSniError(TdsParserStateObject stateObj, UInt32 error)
2024-06-21T15:11:50.9955638Z    at System.Data.SqlClient.TdsParserStateObject.ReadSniSyncOverAsync()
2024-06-21T15:11:50.9957217Z    at System.Data.SqlClient.TdsParserStateObject.TryReadNetworkPacket()
2024-06-21T15:11:50.9957605Z    at System.Data.SqlClient.TdsParserStateObject.TryPrepareBuffer()
2024-06-21T15:11:50.9958017Z    at System.Data.SqlClient.TdsParserStateObject.TryReadByte(Byte& value)
2024-06-21T15:11:50.9958638Z    at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
2024-06-21T15:11:50.9959501Z    at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
2024-06-21T15:11:50.9960461Z    at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
2024-06-21T15:11:50.9961455Z    at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
2024-06-21T15:11:50.9962371Z    at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
2024-06-21T15:11:50.9962957Z    at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
2024-06-21T15:11:50.9963379Z    at bsp.Data.SpExec.ExecCommand(DbConnectionInfo connInfo, String spName, Object[] spParameterList)
2024-06-21T15:11:50.9963838Z    at Project.ImportNG.ServiceData.Save(Int32 recordSessionId, Int32 commitUnit)
2024-06-21T15:11:50.9964284Z    at Project.ImportNG.ImportService.Import(Guid sessionId, CancellationToken cancellationToken)
2024-06-21T15:11:50.9964710Z ClientConnectionId:3befdc8d-7c9b-460b-92c0-fa1b9c9e0ee0
2024-06-21T15:11:50.9965051Z Error Number:64,State:0,Class:20
2024-06-21T15:11:50.9965490Z Project.Console Information: 0 : Updating status of session c711964f-9879-4ef3-9099-e23658db1061 to Exception
2024-06-21T15:11:50.9966702Z Project.Console Error: 0 : Exception during process execution: System.Data.SqlClient.SqlException (0x80131904): A transport-level error has occurred when receiving results from the server. (provider: TCP Provider, error: 0 - The specified network name is no longer available.) ---> System.ComponentModel.Win32Exception (0x80004005): The specified network name is no longer available
2024-06-21T15:11:50.9967743Z    at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
2024-06-21T15:11:50.9968306Z    at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
2024-06-21T15:11:50.9968832Z    at System.Data.SqlClient.TdsParserStateObject.ReadSniError(TdsParserStateObject stateObj, UInt32 error)
2024-06-21T15:11:50.9970333Z    at System.Data.SqlClient.TdsParserStateObject.ReadSniSyncOverAsync()
2024-06-21T15:11:50.9970714Z    at System.Data.SqlClient.TdsParserStateObject.TryReadNetworkPacket()
2024-06-21T15:11:50.9971079Z    at System.Data.SqlClient.TdsParserStateObject.TryPrepareBuffer()
2024-06-21T15:11:50.9971445Z    at System.Data.SqlClient.TdsParserStateObject.TryReadByte(Byte& value)
2024-06-21T15:11:50.9972028Z    at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
2024-06-21T15:11:50.9972821Z    at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
2024-06-21T15:11:50.9974755Z    at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
2024-06-21T15:11:50.9975708Z    at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
2024-06-21T15:11:50.9976554Z    at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
2024-06-21T15:11:50.9977096Z    at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
2024-06-21T15:11:50.9977495Z    at bsp.Data.SpExec.ExecCommand(DbConnectionInfo connInfo, String spName, Object[] spParameterList)
2024-06-21T15:11:50.9977928Z    at Project.ImportNG.ServiceData.Save(Int32 recordSessionId, Int32 commitUnit)
2024-06-21T15:11:50.9978348Z    at Project.ImportNG.ImportService.Import(Guid sessionId, CancellationToken cancellationToken)
2024-06-21T15:11:50.9978797Z    at Project.Console.ServiceRunner.CreateAndRunImportSession(Arguments args, ServiceContext ctx)
2024-06-21T15:11:50.9979151Z    at Project.Console.Program.Execute()
2024-06-21T15:11:50.9979472Z ClientConnectionId:3befdc8d-7c9b-460b-92c0-fa1b9c9e0ee0
2024-06-21T15:11:50.9979786Z Error Number:64,State:0,Class:20
2024-06-21T15:11:50.9980172Z Project.Console Information: 0 : prtConsole tracing finished at 6/21/2024 11:10:24 AM.
2024-06-21T15:11:50.9980598Z Project.Console Information: 0 : Total elapsed time: 00:00:13.796
2024-06-21T15:11:50.9980933Z prtConsole.exe Exit Code = 2

**Comment by arellegue** [comment]
Thank you for reporting this issue.  Before we can begin investigation could please provide the following information:
1. Step how to reproduce the issue.
2. A sample reproduction code.
3. Which version of SqlClient driver and the Sql Server are you using.

Once we receive the above information we can start the investigation.

**Comment by ErikEJ** [comment]
Callstack is System.Data.SqlClient - not supported here

**Comment by kf-gonzalez2** [comment]
Can you try to use Microsoft.Data.SqlClient and let us know if you experience the same issue?

**Comment by JRahnama** [comment]
Closing due to inactivity. Feel free to comment here or open a new issue.
